macosx/PrefsController.m

```diff
@@ -845,7 +845,8 @@ - (void) folderSheetShow: (id) sender
             [fDefaults setBool: YES forKey: @"DownloadLocationConstant"];
             [self updateShowAddMagnetWindowField];
             
-            tr_sessionSetDownloadDir(fHandle, [folder UTF8String]);
+            assert(folder.length > 0);
```

Comment:
I don't think this is a valid use of `assert`. Path validity should be checked on consumer side, and in a way that leads to call failure (not program termination) regardless of whether `NDEBUG` is defined or not.